src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/SystemWebConfigurationProvider.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer;
+
+namespace Microsoft.AspNetCore.Razor.OmniSharpPlugin
+{
+    [Shared]
+    [Export(typeof(RazorConfigurationProvider))]
+    internal class SystemWebConfigurationProvider : CoreProjectConfigurationProvider
+    {
+        // Internal for testing
+        internal const string ReferencePathWithRefAssembliesItemType = "ReferencePathWithRefAssemblies";
+        internal const string SystemWebRazorAssemblyFileName = "System.Web.Razor.dll";
+
+        private const string LegacyRazorAssemblyName = "System.Web.Razor";
+
+        public override bool TryResolveConfiguration(RazorConfigurationProviderContext context, out RazorConfiguration configuration)
+        {
+            if (HasRazorCoreCapability(context))
+            {
+                configuration = null;
+                return false;
+            }
+
+            var compilationReferences = context.ProjectInstance.GetItems(ReferencePathWithRefAssembliesItemType);
+            string legacyRazorReferenceFullPath = null;
+            foreach (var compilationReference in compilationReferences)
+            {
+                var assemblyPath = compilationReference.EvaluatedInclude;
+                if (compilationReference.EvaluatedInclude.EndsWith(SystemWebRazorAssemblyFileName, StringComparison.OrdinalIgnoreCase))
```

Comment:
I don't know if `compilationReference.EvaluatedInclude` is a path string here, but if it is, maybe we need the check to only match if there's a directory separator before `SystemWebRazorAssemblyFileName` (to avoid matching `TaylorSystem.Web.Razor.dll`).

Reply:
Good catch, ya i'll definitely do that.